cobbler/actions/check.py

```diff
@@ -135,7 +135,19 @@ def check_service(self, status, which, notes=""):
         if notes != "":
             notes = " (NOTE: %s)" % notes
         return_code = 0
-        if self.checked_family in ("redhat", "suse"):
+        if utils.is_supervisord():
+            return_code = utils.subprocess_call("supervisorctl status %s > /dev/null 2>/dev/null" % which,
```

Comment:
Another (in my eyes preferred) option would be to use the XML-RPC interface because we don't need to rely on exit codes or stdout parsing.

Docs: http://supervisord.org/api.html

This means we would have more Python Code but the detection would be much more reliable imho.